jira-feature-collector/src/main/java/com/capitalone/dashboard/client/DefaultJiraClient.java

```diff
@@ -0,0 +1,205 @@
+import org.springframework.stereotype.Component;
+
+import com.atlassian.jira.rest.client.api.JiraRestClient;
+import com.atlassian.jira.rest.client.api.RestClientException;
+import com.atlassian.jira.rest.client.api.domain.BasicProject;
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.atlassian.jira.rest.client.api.domain.SearchResult;
+import com.atlassian.util.concurrent.Promise;
+import com.capitalone.dashboard.util.FeatureSettings;
+import com.capitalone.dashboard.util.FeatureWidgetQueries;
+import com.google.common.collect.Lists;
+
+@Component
+public class DefaultJiraClient implements JiraClient {
+	private static final Logger LOGGER = LoggerFactory.getLogger(DefaultJiraClient.class);
+	
+	private final DateFormat QUERY_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+	
+	private static final Set<String> DEFAULT_FIELDS = new HashSet<>();
+	static {
+		DEFAULT_FIELDS.add("*all,-comment,-watches,-worklog,-votes,-reporter,-creator,-attachment");
+	}
+	
+	private final FeatureSettings featureSettings;
+	private final FeatureWidgetQueries featureWidgetQueries;
+	
+	private JiraRestClient client;
+	
+	@Autowired
+	public DefaultJiraClient(FeatureSettings featureSettings, FeatureWidgetQueries featureWidgetQueries, JiraRestClientSupplier restSupplier) {
+		this.featureSettings = featureSettings;
+		this.featureWidgetQueries = featureWidgetQueries;
+		this.client = restSupplier.get();
+	}
+	
+	@Override
+	public List<Issue> getIssues(long startTime, int pageStart) {
+		List<Issue> rt = new ArrayList<>();
+		
+		if (client != null) {
+			try {
+				// example "1900-01-01 00:00"
+				String startDateStr = QUERY_DATE_FORMAT.format(new Date(startTime));
+				
+				String query = featureWidgetQueries.getStoryQuery(startDateStr,
+						featureSettings.getJiraIssueTypeId(), featureSettings.getStoryQuery());
+				
+				Promise<SearchResult> promisedRs = client.getSearchClient().searchJql(
+						query, featureSettings.getPageSize(), pageStart, DEFAULT_FIELDS);
+				
+				SearchResult sr = promisedRs.claim();
+
+				Iterable<Issue> jiraRawRs = sr.getIssues();
+				
+				if (jiraRawRs != null) {
+					if (LOGGER.isDebugEnabled()) {
+						int pageEnd = Math.min(pageStart + getPageSize() - 1, sr.getTotal());
+						
+						LOGGER.debug(String.format("Processing issues %d - %d out of %d", pageStart, pageEnd, sr.getTotal()));
+					}
+					
+					rt = Lists.newArrayList(jiraRawRs);
+				}
+			} catch (RestClientException e) {
+				if (e.getStatusCode().get() != null && e.getStatusCode().get() == 401 ) {
+					LOGGER.error("Error 401 connecting to JIRA server, your credentials are probably wrong. Note: Ensure you are using JIRA user name not your email address.");
+				} else {
+					LOGGER.error("No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:" + e.getCause());
+				}
+				if (LOGGER.isDebugEnabled()) {

```

Comment:
Its hard to read on my phone can u confirm we r not suppressing the 401 error

Sent from my iPhone

On Jul 7, 2016, at 08:07, Mark <notifications@github.com<mailto:notifications@github.com>> wrote:


In jira-feature-collector/src/main/java/com/capitalone/dashboard/client/DefaultJiraClient.java<https://github.com/capitalone/Hygieia/pull/675#discussion_r69924061>:

> +                             if (jiraRawRs != null) {
> +                                     if (LOGGER.isDebugEnabled()) {
> +                                             int pageEnd = Math.min(pageStart + getPageSize() - 1, sr.getTotal());
> +
> +                                             LOGGER.debug(String.format("Processing issues %d - %d out of %d", pageStart, pageEnd, sr.getTotal()));
> +                                     }
> +
> +                                     rt = Lists.newArrayList(jiraRawRs);
> +                             }
> +                     } catch (RestClientException e) {
> +                             if (e.getStatusCode().get() != null && e.getStatusCode().get() == 401 ) {
> +                                     LOGGER.error("Error 401 connecting to JIRA server, your credentials are probably wrong. Note: Ensure you are using JIRA user name not your email address.");
> +                             } else {
> +                                     LOGGER.error("No result was available from Jira unexpectedly - defaulting to blank response. The reason for this fault is the following:" + e.getCause());
> +                             }
> +                             if (LOGGER.isDebugEnabled()) {


It's a practice to prevent debug code from interfering / modifying production behavior. Been bitten by that before. This way if there was a bug in the debug code (such as creating an exception) it could be avoided by turning off debug without having to recompile.

-
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/capitalone/Hygieia/pull/675/files/af67431db6b3aa308db5e334aeea223cf02d7d3a#r69924061>, or mute the thread<https://github.com/notifications/unsubscribe/ANtPT1L9lwjGAF4vpwkL1D8TJxjRrxNJks5qTRYXgaJpZM4JGQCE>.